### ISSUE TITLE
nginx_uwsgi_test intermittently failing

### ISSUE THREAD

**Issue by rorour** [radiasoft/sirepo#4554]
nginx_uwsgi_test "randomly" fails. In the github ci environment, this is the log.

```
============================= test session starts ==============================
platform linux -- Python 3.7.2, pytest-7.1.2, pluggy-1.0.0 -- /home/vagrant/.pyenv/versions/3.7.2/envs/py3/bin/python
cachedir: .pytest_cache
rootdir: /home/runner/work/sirepo/sirepo
collecting ... collected 1 item
tests/nginx_uwsgi_test.py::test_ping Jun 23 18:05:34 10585     0 ../../sirepo/job_driver/__init__.py:396:init modules=['local']
Jun 23 18:05:34 10585     0 ../../sirepo/pkcli/job_supervisor.py:77:default_command ip=127.0.0.1 port=8101
nginx: [alert] could not open error log file: open() "/var/log/nginx/error.log" failed (13: Permission denied)
2022/06/23 18:05:34 [warn] 10594#10594: could not build optimal types_hash, you should increase either types_hash_max_size: 2048 or types_hash_bucket_size: 64; ignoring types_hash_bucket_size
[uWSGI] getting YAML configuration from /home/runner/work/sirepo/sirepo/tests/nginx_uwsgi_work/db/uwsgi.yml
*** Starting uWSGI 2.0.20 (64bit) on [Thu Jun 23 18:05:34 2022] ***
compiled with version: 10.3.1 20210422 (Red Hat 10.3.1-1) on 23 June 2022 17:59:05
os: Linux-5.13.0-1031-azure #37~20.04.1-Ubuntu SMP Mon Jun 13 22:51:01 UTC 2022
nodename: b5889dfaefd9
machine: x86_64
clock source: unix
pcre jit disabled
detected number of CPU cores: 2
current working directory: /home/runner/work/sirepo/sirepo/tests/nginx_uwsgi_work/db
writing pidfile to /home/runner/work/sirepo/sirepo/tests/nginx_uwsgi_work/db/uwsgi.pid
detected binary path: /home/vagrant/.pyenv/versions/3.7.2/envs/py3/bin/uwsgi
your memory page size is 4096 bytes
detected max file descriptor number: 1048576
lock engine: pthread robust mutexes
thunder lock: disabled (you can enable it with --thunder-lock)
uwsgi socket 0 bound to TCP address 0.0.0.0:8102 fd 3
Python version: 3.7.2 (default, May 26 2022, 15:28:25)  [GCC 10.3.1 20210422 (Red Hat 10.3.1-1)]
Python main interpreter initialized at 0x1450d00
python threads support enabled
your server socket listen backlog is limited to 100 connections
your mercy for graceful operations on workers is 60 seconds
mapped 33[404](https://github.com/radiasoft/sirepo/runs/7029182473?check_suite_focus=true#step:3:405)8 bytes (326 KB) for 10 cores
*** Operational MODE: threaded ***
Jun 23 18:05:35 10596     0 ../../../sirepo/uri_router.py:199:register_api_module module=<module 'sirepo.auth.basic' from '/home/runner/work/sirepo/sirepo/sirepo/auth/basic.py'> does not have API class; no apis
Jun 23 18:05:35 10596     0 ../../../sirepo/db_upgrade.py:36:do_all running upgrade _20210211_add_flash_proprietary_lib_files
Jun 23 18:05:35 10596     0 ../../../sirepo/db_upgrade.py:36:do_all running upgrade _20210211_upgrade_runner_to_job_db
Jun 23 18:05:35 10596     0 ../../../sirepo/db_upgrade.py:36:do_all running upgrade _20210218_add_flash_proprietary_lib_files_force
Jun 23 18:05:35 10596     0 ../../../sirepo/db_upgrade.py:36:do_all running upgrade _20210301_migrate_role_jupyterhub
Jun 23 18:05:35 10596     0 ../../../sirepo/db_upgrade.py:36:do_all running upgrade _20220609_add_expiration_column_to_user_role_t
WSGI app 0 (mountpoint='') ready in 1 seconds on interpreter 0x1450d00 pid: 10596 (default app)
*** uWSGI is running in multiple interpreter mode ***
spawned uWSGI master process (pid: 10596)
spawned uWSGI worker 1 (pid: 10600, cores: 10)
*** Stats server enabled on /home/runner/work/sirepo/sirepo/tests/nginx_uwsgi_work/db/uwsgi.sock fd: 9 ***
ERROR
==================================== ERRORS ====================================
_________________________ ERROR at setup of test_ping __________________________
Traceback (most recent call last):
  File "/home/runner/work/sirepo/sirepo/tests/conftest.py", line 361, in _subprocess_start
    r = c.sr_post('jobSupervisorPing', PKDict(simulationType=sirepo.srunit.SR_SIM_TYPE_DEFAULT))
  File "/home/runner/work/sirepo/sirepo/sirepo/srunit.py", line 179, in sr_post
    r.raise_for_status()
  File "/home/vagrant/.pyenv/versions/3.7.2/envs/py3/lib/python3.7/site-packages/requests/models.py", line 960, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 502 Server Error: Bad Gateway for url: http://0.0.0.0:8180/job-supervisor-ping
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "/home/runner/work/sirepo/sirepo/tests/conftest.py", line 173, in uwsgi_module
    with _auth_client_module(request, uwsgi=True) as c:
  File "/home/vagrant/.pyenv/versions/3.7.2/lib/python3.7/contextlib.py", line 112, in __enter__
    return next(self.gen)
  File "/home/runner/work/sirepo/sirepo/tests/conftest.py", line 204, in _auth_client_module
    with _subprocess_start(request, cfg=cfg, uwsgi=uwsgi) as c:
  File "/home/vagrant/.pyenv/versions/3.7.2/lib/python3.7/contextlib.py", line 112, in __enter__
    return next(self.gen)
  File "/home/runner/work/sirepo/sirepo/tests/conftest.py", line 376, in _subprocess_start
    x.wait()
  File "/home/vagrant/.pyenv/versions/3.7.2/lib/python3.7/subprocess.py", line 990, in wait
    return self._wait(timeout=timeout)
  File "/home/vagrant/.pyenv/versions/3.7.2/lib/python3.7/subprocess.py", line 1624, in _wait
    (pid, sts) = self._try_wait(0)
  File "/home/vagrant/.pyenv/versions/3.7.2/lib/python3.7/subprocess.py", line 1582, in _try_wait
    (pid, sts) = os.waitpid(self.pid, wait_flags)
  File "/home/runner/work/sirepo/sirepo/tests/conftest.py", line 137, in _timeout
    pkunit.pkfail('MAX_CASE_RUN_SECS={} exceeded', MAX_CASE_RUN_SECS)
  File "/home/vagrant/.pyenv/versions/3.7.2/envs/py3/lib/python3.7/site-packages/pykern/pkunit.py", line 351, in pkfail
    raise PKFail('{} {}'.format(call, msg))
pykern.pkunit.PKFail: tests/conftest.py:137:_timeout MAX_CASE_RUN_SECS=120 exceeded
```

**Comment by rorour** [comment]
Another common error:

```
==================================== ERRORS ====================================
_________________________ ERROR at setup of test_ping __________________________
Traceback (most recent call last):
  File "/home/vagrant/src/radiasoft/sirepo/tests/conftest.py", line 173, in uwsgi_module
    with _auth_client_module(request, uwsgi=True) as c:
  File "/home/vagrant/.pyenv/versions/3.7.2/lib/python3.7/contextlib.py", line 112, in __enter__
    return next(self.gen)
  File "/home/vagrant/src/radiasoft/sirepo/tests/conftest.py", line 204, in _auth_client_module
    with _subprocess_start(request, cfg=cfg, uwsgi=uwsgi) as c:
  File "/home/vagrant/.pyenv/versions/3.7.2/lib/python3.7/contextlib.py", line 112, in __enter__
    return next(self.gen)
  File "/home/vagrant/src/radiasoft/sirepo/tests/conftest.py", line 371, in _subprocess_start
    pkunit.pkfail('could not connect to {}', sirepo.job_api.cfg.supervisor_uri)
  File "/home/vagrant/src/radiasoft/pykern/pykern/pkunit.py", line 342, in pkfail
    raise PKFail('{} {}'.format(call, msg))
pykern.pkunit.PKFail: tests/conftest.py:371:_subprocess_start could not connect to http://127.0.0.1:8001
=============================== 1 error in 5.96s ===============================

```

**Comment by rorour** [comment]
Another common error:

```
==================================== ERRORS ====================================
_________________________ ERROR at setup of test_ping __________________________
Traceback (most recent call last):
  File "/home/vagrant/src/radiasoft/sirepo/tests/conftest.py", line 244, in _job_supervisor_check
    s.bind((env.SIREPO_PKCLI_JOB_SUPERVISOR_IP, int(env.SIREPO_PKCLI_JOB_SUPERVISOR_PORT)))
OSError: [Errno 98] Address already in use

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/vagrant/src/radiasoft/sirepo/tests/conftest.py", line 173, in uwsgi_module
    with _auth_client_module(request, uwsgi=True) as c:
  File "/home/vagrant/.pyenv/versions/3.7.2/lib/python3.7/contextlib.py", line 112, in __enter__
    return next(self.gen)
  File "/home/vagrant/src/radiasoft/sirepo/tests/conftest.py", line 204, in _auth_client_module
    with _subprocess_start(request, cfg=cfg, uwsgi=uwsgi) as c:
  File "/home/vagrant/.pyenv/versions/3.7.2/lib/python3.7/contextlib.py", line 112, in __enter__
    return next(self.gen)
  File "/home/vagrant/src/radiasoft/sirepo/tests/conftest.py", line 350, in _subprocess_start
    env, c = _subprocess_setup(request, cfg, uwsgi)
  File "/home/vagrant/src/radiasoft/sirepo/tests/conftest.py", line 336, in _subprocess_setup
    _job_supervisor_check(env)
  File "/home/vagrant/src/radiasoft/sirepo/tests/conftest.py", line 249, in _job_supervisor_check
    env.SIREPO_PKCLI_JOB_SUPERVISOR_PORT,
AssertionError: job_supervisor still running on ip=127.0.0.1 port=8101
=============================== 1 error in 0.43s ===============================
error: FAILED=1 passed=0

```

**Comment by rorour** [comment]
this seems to frequently be the cause of `pykern.pkunit.PKFail: tests/conftest.py:371:_subprocess_start could not connect to http://127.0.0.1:8001
`

```
============================= test session starts ==============================
platform linux -- Python 3.7.2, pytest-7.1.2, pluggy-1.0.0 -- /home/vagrant/.pyenv/versions/3.7.2/envs/py3/bin/python
cachedir: .pytest_cache
rootdir: /home/vagrant/src/radiasoft/sirepo
plugins: anyio-3.5.0
collecting ... collected 1 item

tests/nginx_uwsgi_test.py::test_ping Traceback (most recent call last):
  File "/home/vagrant/.pyenv/versions/3.7.2/envs/py3/lib/python3.7/site-packages/sqlalchemy/engine/base.py", line 1803, in _execute_context
    cursor, statement, parameters, context
  File "/home/vagrant/.pyenv/versions/3.7.2/envs/py3/lib/python3.7/site-packages/sqlalchemy/engine/default.py", line 732, in do_execute
    cursor.execute(statement, parameters)
sqlite3.OperationalError: table db_upgrade_t already exists

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/vagrant/.pyenv/versions/py3/bin/sirepo", line 33, in <module>
    sys.exit(load_entry_point('sirepo', 'console_scripts', 'sirepo')())
  File "/home/vagrant/src/radiasoft/sirepo/sirepo/sirepo_console.py", line 18, in main
    return pkcli.main('sirepo')
  File "/home/vagrant/src/radiasoft/pykern/pykern/pkcli/__init__.py", line 143, in main
    res = argh.dispatch(parser, argv=argv)
  File "/home/vagrant/.pyenv/versions/3.7.2/envs/py3/lib/python3.7/site-packages/argh/dispatching.py", line 174, in dispatch
    for line in lines:
  File "/home/vagrant/.pyenv/versions/3.7.2/envs/py3/lib/python3.7/site-packages/argh/dispatching.py", line 277, in _execute_command
    for line in result:
  File "/home/vagrant/.pyenv/versions/3.7.2/envs/py3/lib/python3.7/site-packages/argh/dispatching.py", line 260, in _call
    result = function(*positional, **keywords)
  File "/home/vagrant/src/radiasoft/sirepo/sirepo/pkcli/service.py", line 157, in nginx_proxy
    run_dir = _run_dir().join('nginx_proxy').ensure(dir=True)
  File "/home/vagrant/src/radiasoft/sirepo/sirepo/pkcli/service.py", line 258, in _run_dir
    from sirepo import server
  File "/home/vagrant/src/radiasoft/sirepo/sirepo/server.py", line 15, in <module>
    from sirepo import simulation_db
  File "/home/vagrant/src/radiasoft/sirepo/sirepo/simulation_db.py", line 29, in <module>
    import sirepo.auth
  File "/home/vagrant/src/radiasoft/sirepo/sirepo/auth/__init__.py", line 797, in <module>
    _init()
  File "/home/vagrant/src/radiasoft/sirepo/sirepo/auth/__init__.py", line 667, in _init
    _init_full()
  File "/home/vagrant/src/radiasoft/sirepo/sirepo/auth/__init__.py", line 672, in _init_full
    auth_db.init()
  File "/home/vagrant/src/radiasoft/sirepo/sirepo/auth_db.py", line 402, in init
    _create_tables(_engine)
  File "/home/vagrant/src/radiasoft/sirepo/sirepo/auth_db.py", line 125, in _create_tables
    b.metadata.create_all(engine)
  File "/home/vagrant/.pyenv/versions/3.7.2/envs/py3/lib/python3.7/site-packages/sqlalchemy/sql/schema.py", line 4786, in create_all
    ddl.SchemaGenerator, self, checkfirst=checkfirst, tables=tables
  File "/home/vagrant/.pyenv/versions/3.7.2/envs/py3/lib/python3.7/site-packages/sqlalchemy/engine/base.py", line 3117, in _run_ddl_visitor
    conn._run_ddl_visitor(visitorcallable, element, **kwargs)
  File "/home/vagrant/.pyenv/versions/3.7.2/envs/py3/lib/python3.7/site-packages/sqlalchemy/engine/base.py", line 2113, in _run_ddl_visitor
    visitorcallable(self.dialect, self, **kwargs).traverse_single(element)
  File "/home/vagrant/.pyenv/versions/3.7.2/envs/py3/lib/python3.7/site-packages/sqlalchemy/sql/visitors.py", line 524, in traverse_single
    return meth(obj, **kw)
  File "/home/vagrant/.pyenv/versions/3.7.2/envs/py3/lib/python3.7/site-packages/sqlalchemy/sql/ddl.py", line 853, in visit_metadata
    _is_metadata_operation=True,
  File "/home/vagrant/.pyenv/versions/3.7.2/envs/py3/lib/python3.7/site-packages/sqlalchemy/sql/visitors.py", line 524, in traverse_single
    return meth(obj, **kw)
  File "/home/vagrant/.pyenv/versions/3.7.2/envs/py3/lib/python3.7/site-packages/sqlalchemy/sql/ddl.py", line 898, in visit_table
    include_foreign_key_constraints,  # noqa
  File "/home/vagrant/.pyenv/versions/3.7.2/envs/py3/lib/python3.7/site-packages/sqlalchemy/engine/base.py", line 1289, in execute
    return meth(self, multiparams, params, _EMPTY_EXECUTION_OPTS)
  File "/home/vagrant/.pyenv/versions/3.7.2/envs/py3/lib/python3.7/site-packages/sqlalchemy/sql/ddl.py", line 81, in _execute_on_connection
    self, multiparams, params, execution_options
  File "/home/vagrant/.pyenv/versions/3.7.2/envs/py3/lib/python3.7/site-packages/sqlalchemy/engine/base.py", line 1387, in _execute_ddl
    compiled,
  File "/home/vagrant/.pyenv/versions/3.7.2/envs/py3/lib/python3.7/site-packages/sqlalchemy/engine/base.py", line 1846, in _execute_context
    e, statement, parameters, cursor, context
  File "/home/vagrant/.pyenv/versions/3.7.2/envs/py3/lib/python3.7/site-packages/sqlalchemy/engine/base.py", line 2027, in _handle_dbapi_exception
    sqlalchemy_exception, with_traceback=exc_info[2], from_=e
  File "/home/vagrant/.pyenv/versions/3.7.2/envs/py3/lib/python3.7/site-packages/sqlalchemy/util/compat.py", line 207, in raise_
    raise exception
  File "/home/vagrant/.pyenv/versions/3.7.2/envs/py3/lib/python3.7/site-packages/sqlalchemy/engine/base.py", line 1803, in _execute_context
    cursor, statement, parameters, context
  File "/home/vagrant/.pyenv/versions/3.7.2/envs/py3/lib/python3.7/site-packages/sqlalchemy/engine/default.py", line 732, in do_execute
    cursor.execute(statement, parameters)
sqlalchemy.exc.OperationalError: (sqlite3.OperationalError) table db_upgrade_t already exists
[SQL: 
CREATE TABLE db_upgrade_t (
	name VARCHAR(100) NOT NULL, 
	created DATETIME NOT NULL, 
	PRIMARY KEY (name)
)

]
(Background on this error at: https://sqlalche.me/e/14/e3q8)
```

**Comment by robnagler** [comment]
I think this is the error:

```py
tests/nginx_uwsgi_test.py::test_ping Traceback (most recent call last):
  File "/home/vagrant/.pyenv/versions/3.7.2/envs/py3/lib/python3.7/site-packages/sqlalchemy/engine/base.py", line 1803, in _execute_context
    cursor, statement, parameters, context
  File "/home/vagrant/.pyenv/versions/3.7.2/envs/py3/lib/python3.7/site-packages/sqlalchemy/engine/default.py", line 732, in do_execute
    cursor.execute(statement, parameters)
sqlite3.OperationalError: table db_upgrade_t already exists
```

There's a race condition when the server and supervisor start. The test code that starts them needs to be more robust, checking that the supervisor has created the db and is ready to go before starting the server.

**Comment by robnagler** [comment]
Fixed by #4570
